lib/daru/index.rb

```diff
@@ -357,4 +406,106 @@ def conform input_indexes
       drop_left_level input_indexes.size
     end
   end
-end
+
+  class CategoricalIndex < Index
+    
+    include Enumerable
+    
+    attr_reader :cat_hash, :array, :map_cat_int, :map_int_cat
+    
+    def initialize indexes
+      # Create a hash to map each category to positional indexes
+      @cat_hash = Hash.new []
+      # Create an array to map each positional index to its category
```

Comment:
`@array` does not contain the actual categories, but rather the integers representing each category, right? So, I found this comment a little misleading.